File: DSA/DSA.Algorithms/SearchingAlgorithmsNS/SearchingAlgorithms.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSA.Algorithms.SearchingAlgorithmsNS
{
    public static class SearchingAlgorithms
    {
        public static int? BinarySearch(int[] arr, int target)
        {
            int left = 0, right = arr.Length - 1;

            while (left <= right)
            {
                int mid = (right + left) / 2;

                if (arr[mid] == target) return mid;

                if (arr[mid] > target) right = mid - 1;
                if (arr[mid] < target) left = mid + 1;
            }
            return null;
        }
    }
}

File: DSA/src/algorithms/searching/SearchingAlgorithms.java
package algorithms.searching;

public class SearchingAlgorithms {
    public static int binarySearch(int[] arr, int target) {
        int left = 0, right = arr.length - 1, mid = right / 2;

        while (left < arr.length && right >= 0) {
            if (arr[mid] == target) return mid;
            if (arr[mid] > target) {
                right = mid - 1;
                mid = (right + left) / 2;
                continue;
            }
            if (arr[mid] < target) {
                left = mid + 1;
                mid = (right + left) / 2;
            }
        }
        return -1;
    }
}
